sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/SyncAccountRepository.kt

```diff
@@ -234,6 +393,85 @@ class AppSyncAccountRepository @Inject constructor(
         }
     }
 
+    override fun pollSecondDeviceExchangeAcknowledgement(): Result<Boolean> {
+        // Sync: InviteFlow - C (https://app.asana.com/0/72649045549333/1209571867429615)
+        Timber.d("Sync-exchange: InviteFlow - C")
+
+        val keyId = pendingInvitationAsInviter?.keyId ?: return Error(reason = "No pending invitation initialized")
+
+        return when (val result = syncApi.getEncryptedMessage(keyId)) {
+            is Error -> {
+                if (result.code == NOT_FOUND.code) {
+                    return Success(false)
+                } else if (result.code == GONE.code) {
+                    return Error(
+                        code = CONNECT_FAILED.code,
+                        reason = "Connect: keys expired",
+                    ).alsoFireAccountErrorPixel()
+                }
+                result.alsoFireAccountErrorPixel()
+            }
+
+            is Success -> {
+                Timber.v("Sync-exchange: Found invitation acceptance for keyId: $keyId} ${result.data}")
+
+                val decrypted = kotlin.runCatching {
+                    val pending = pendingInvitationAsInviter
+                        ?: return Error(code = CONNECT_FAILED.code, reason = "Exchange: No pending invitation initialized")
+                            .alsoFireAccountErrorPixel()
+
+                    nativeLib.sealOpen(result.data, pending.publicKey, pending.privateKey)
+                }.getOrElse { throwable ->
+                    throwable.asErrorResult().alsoFireAccountErrorPixel()
+                    return Error(code = CONNECT_FAILED.code, reason = "Connect: Error opening seal")
+                }
+
+                Timber.v("Sync-exchange: invitation acceptance received: $decrypted")
+
+                val response = Adapters.invitedDeviceAdapter.fromJson(decrypted)
+                    ?: return Error(code = GENERIC_ERROR.code, reason = "Connect: Error reading invitation response").alsoFireAccountErrorPixel()
+
+                val otherDevicePublicKey = response.publicKey
+                val otherDeviceKeyId = response.keyId
+
+                Timber.v(
+                    "Sync-exchange: We have received the other device's details. " +
+                        "name:${response.deviceName}, keyId:${response.keyId}, public key: ${response.publicKey}",
+                )
+
+                // we encrypt our secrets with otherDevicePublicKey, and send them to the backend endpoint
+                return sendSecrets(otherDeviceKeyId, otherDevicePublicKey).onFailure {
+                    Timber.w("Sync-exchange: failed to send secrets. error code: ${it.code} ${it.reason}")
+                    return it.copy(code = LOGIN_FAILED.code)
```

Comment:
This should be exchange failed right? sending secrets we are not performing login